ceph.spec.in

```diff
@@ -947,7 +947,7 @@ Summary:	Ceph distributed file system client library
 %if 0%{?suse_version}
 Group:		System/Libraries
 %endif
-Obsoletes:	libcephfs1
+Obsoletes:	libcephfs1 < %{_epoch_prefix}%{version}-%{release}
 %if 0%{?rhel} || 0%{?fedora}
 Obsoletes:	ceph-libs < %{_epoch_prefix}%{version}-%{release}
```

Comment:
@tchaikov I deliberately chose to be "non-specific" for two reasons:

1. using `Obsoletes: libcephfs1 < %{_epoch_prefix}%{version}-%{release}` is as close as I can come to the meaning of the previous version `Obsoletes: libcephfs1` (i.e. "libcephfs1 is absolutely obsolete and should not be used, regardless of version number") without triggering a warning
2. I like to be consistent and we are using this formulation elsewhere in the spec file, e.g. here in line 952.

Reply:
Yes, it is a common practice to obsolete the packages to anything less than the current version to allow possible future re-appearences of the package.

The theory basically goes as follows: 

You can only control dependencies of the current build and you know that the current build wouldn't play well with an older build that would contain the package. There could be some older (custom) build that contained the package version and this build could be newer than the git log suggests.

-> Your safest bet is usually to obsolete the package in any older version than you are building right now.